### PR TITLE
Delay sorting of evictUnits slice in kuberuntime_gc

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Currently the sorting of evictUnits slices is done in cgc.evictableContainers

This PR delays the sorting to when the sorting is needed in enforceMaxContainersPerEvictUnit

Same approach is applied to removing sandboxes.

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_gc.go [hunk 1]
```diff
@@ -116,6 +116,9 @@ func (cgc *containerGC) enforceMaxContainersPerEvictUnit(evictUnits containersBy
 		toRemove := len(evictUnits[key]) - MaxContainers
 
 		if toRemove > 0 {
+			if len(evictUnits[key]) > toRemove {
```
Comment (mattjmcnaughton) [inline pkg/kubelet/kuberuntime/kuberuntime_gc.go]: Could we instead move this to `removeOldestN` similar to `removeOldestNSandboxes`?

**Review by mattjmcnaughton — CHANGES_REQUESTED** [review]
Overall, looks good to me! Just requiring one small readability change.

**Comment by tedyu** [comment]
@yujuhong @tallclair 
Please review.

**Comment by tedyu** [comment]
@derekwaynecarr 
Please review

**Comment by tedyu** [comment]
@derekwaynecarr
Please review

**Comment by tedyu** [comment]
cc @vishh

**Comment by tedyu** [comment]
@vishh
Please review.

**Comment by tedyu** [comment]
cc @mattjmcnaughton

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

**Comment by tedyu** [comment]
ping @dchen1107 for review

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
@vishh
Please review.

**Comment by tedyu** [comment]
@vishh
Please review.

**Comment by tedyu** [comment]
ping @derekwaynecarr for review

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
ping @yujuhong for review.

**Comment by tedyu** [comment]
ping @vishh for review

**Comment by tedyu** [comment]
@vishh
please review

**Comment by tedyu** [comment]
@vishh
please review

**Comment by tedyu** [comment]
@vishh
please review

**Comment by tedyu** [comment]
@dchen1107
Can you take a look ?

**Pull request closed without merging**